src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/SuppressedIssuesFileSynchronizer.cs

```diff
@@ -53,34 +57,59 @@ public IEnumerable<SonarQubeIssue> Get(string sonarProjectKey)
     {
+        private readonly IThreadHandling threadHandling;
         private readonly ISuppressedIssuesMonitor suppressedIssuesMonitor;
         private readonly ISonarQubeIssuesProvider suppressedIssuesProvider;
         private readonly ISuppressedIssuesFileStorage suppressedIssuesFileStorage;
         private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
 
         [ImportingConstructor]
-        public SuppressedIssuesFileSynchronizer(ISuppressedIssuesMonitor suppressedIssuesMonitor, 
+        public SuppressedIssuesFileSynchronizer(ISuppressedIssuesMonitor suppressedIssuesMonitor,
             ISonarQubeIssuesProvider suppressedIssuesProvider,
             ISuppressedIssuesFileStorage suppressedIssuesFileStorage,
             IActiveSolutionBoundTracker activeSolutionBoundTracker)
+            : this(suppressedIssuesMonitor,
+                suppressedIssuesProvider,
+                suppressedIssuesFileStorage,
+                activeSolutionBoundTracker,
+                new ThreadHandling())
+        {
+        }
+
+        internal SuppressedIssuesFileSynchronizer(ISuppressedIssuesMonitor suppressedIssuesMonitor,
+            ISonarQubeIssuesProvider suppressedIssuesProvider,
+            ISuppressedIssuesFileStorage suppressedIssuesFileStorage,
+            IActiveSolutionBoundTracker activeSolutionBoundTracker,
+            IThreadHandling threadHandling)
         {
             this.suppressedIssuesMonitor = suppressedIssuesMonitor;
             this.suppressedIssuesProvider = suppressedIssuesProvider;
             this.suppressedIssuesFileStorage = suppressedIssuesFileStorage;
             this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.threadHandling = threadHandling;
 
             suppressedIssuesMonitor.SuppressionsUpdateRequested += SuppressedIssuesMonitor_SuppressionsUpdateRequested;
         }
 
         private void SuppressedIssuesMonitor_SuppressionsUpdateRequested(object sender, EventArgs e)
         {
-            var sonarProjectKey = activeSolutionBoundTracker.CurrentConfiguration.Project?.ProjectKey;
+            UpdateFileStorage();
+        }
 
-            if (!string.IsNullOrEmpty(sonarProjectKey))
+        public void UpdateFileStorage()
+        {
+            threadHandling.Run(() =>
             {
-                var allSuppressedIssues = suppressedIssuesProvider.GetAllSuppressedIssues();
-                suppressedIssuesFileStorage.Update(sonarProjectKey, allSuppressedIssues);
-            }
+                var sonarProjectKey = activeSolutionBoundTracker.CurrentConfiguration.Project?.ProjectKey;
+
+                if (!string.IsNullOrEmpty(sonarProjectKey))
+                {
+                    var allSuppressedIssues = suppressedIssuesProvider.GetAllSuppressedIssues();
+                    suppressedIssuesFileStorage.Update(sonarProjectKey, allSuppressedIssues);
+                }
+
+                return Task.FromResult(true);
```

Comment:
A little bit ugly but the current Run method expects a Task<T> and doesn't support void. I wanted to avoid further changes so I'm just returning a bool.